.JuliaFormatter.toml

```diff
@@ -1,3 +1,4 @@
 remove_extra_newlines = true
 always_for_in = true
 conditional_to_if = true
+ignored = ["bin", "test/files", "docs"]
```

Comment:
why is test/files ignored? This is part of the test suite

Reply:
They are unformatted on purpose, right? So `format(JuliaFormatter)` shouldn't be allowed to format them. Note that there's a config file in `test/files`, so `format("test/files")` doesn't read the top level config.

Reply:
They are formatted though, they're part of the test suite when format(".") is called at the top level.

Reply:
just to clarify - does the `ignored` functionality stop if the subdirectory has it's own format file? So if at the top level I have ignored=["subdir"] but then subdir has it's own .JuliaFormatter.toml. Does subdir still get formatted according to its own rules?

Reply:
So that's `test/format_repo.jl`? I guess one could overwrite the `.JuliaFormatter.toml` for that test specifically.

Reply:
Strangely, if I add
```
open(joinpath(sandbox_dir, ".JuliaFormatter.toml"), "w") do io
    write(io, """
          remove_extra_newlines = true
          always_for_in = true
          conditional_to_if = true
          """)
end
```
to that test, the first two `format` calls return `false`, because some comments in `test/files/Docs.jl` are indented, and `test/files/reverse.jl` has a couple of lines being split into several. I don't quite understand why this happens.

Reply:
hmmm I think that makes sense because test/files has it's own .JuliaFormatter.toml file

https://github.com/domluna/JuliaFormatter.jl/blob/master/test/files/.JuliaFormatter.toml

that only has remove_extra_newlines in it so when format(".") is called at the top level only that option will be overwritten (from the defaults)

where as the top level has https://github.com/domluna/JuliaFormatter.jl/blob/master/.JuliaFormatter.toml

```
remove_extra_newlines = true
always_for_in = true
conditional_to_if = true
```

So if you call format on test/files with the options of the top-level then it makes sense some files would turn out differently.

Reply:
The most specific .JuliaFormatter.toml file is chosen. If the structure is

A
  B
  C
   D

and A, C, and D have their own .JuliaFormatter.toml files then each is used for that repo. Files in D uses D's .toml file, same with files in C.

B however since it doesn't have its own file will default to using A's .toml file

Reply:
But then I would expect the second `format` to return `true`, because I'm running the same call twice?

And is the intention for nested configs to overwrite entirely? Because in the current implementation of this PR, if you call `format("A")`, only the options mentioned in `C/.JuliaFormatter.toml` are overwritten - for instance, say `A/.JuliaFormatter.toml` has 
```toml
remove_extra_newlines = true
always_for_in = true
```
 and `C/.JuliaFormatter.toml` has
```toml
always_for_in = false
conditional_to_if = true
```
then the effective config for files in `C` is
```toml
remove_extra_newlines = true
always_for_in = false
conditional_to_if = true
```

I think that's how it worked before the PR too.

Reply:
(This, btw, only applies for `format("A")`, if we run `format("C")`, `A/.JuliaFormatter.toml` is never read, so the effective config is
```
always_for_in = false
conditional_to_if = true
```
)